benchmark/inference/inference_benchmark.py

```diff
@@ -0,0 +1,126 @@
+from torch_geometric.nn import PNAConv
+
+supported_sets = {
+    'ogbn-mag': ['rgat', 'rgcn'],
+    'reddit': ['edge_conv', 'gat', 'gcn', 'pna_conv'],
+    'ogbn-products': ['edge_conv', 'gat', 'gcn', 'pna_conv'],
+}
+
+
+def run(args: argparse.ArgumentParser) -> None:
+
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    progress_bar = True
+
+    print('BENCHMARK STARTS')
+    for dataset_name in args.datasets:
+        print(f'Dataset: {dataset_name}')
+        dataset = get_dataset(dataset_name, args.root)
+
+        hetero = True if dataset_name == 'ogbn-mag' else False
+        mask = ('paper', None) if hetero else None
+        degree = None
+
+        data = dataset[0].to(device)
+        inputs_channels = data.x_dict['paper'].size(
+            -1) if hetero else dataset.num_features
+
+        for model_name in args.models:
+            if model_name not in supported_sets[dataset_name]:
+                print(f'Configuration of {dataset_name} + {model_name} '
+                      f'not supported. Skipping.')
+                continue
+            print(f'Evaluation bench for {model_name}:')
+
+            for batch_size in args.eval_batch_sizes:
+                if not hetero:
+                    subgraph_loader = NeighborLoader(
+                        copy.copy(data),
+                        num_neighbors=[-1],
+                        input_nodes=mask,
+                        batch_size=batch_size,
+                        shuffle=False,
+                        num_workers=args.num_workers,
+                    )
+                    subgraph_loader.data.n_id = torch.arange(data.num_nodes)
+
+                for layers in args.num_layers:
+                    if hetero:
+                        subgraph_loader = NeighborLoader(
+                            copy.copy(data),
+                            num_neighbors=[args.hetero_num_neighbors] * layers,
+                            input_nodes=mask,
+                            batch_size=batch_size,
+                            shuffle=False,
+                            num_workers=args.num_workers,
+                        )
+                        subgraph_loader.data.n_id = torch.arange(
+                            data.num_nodes)
+
+                    for hidden_channels in args.num_hidden_channels:
+                        print(
+                            '-----------------------------------------------')
+                        print(f'Batch size={batch_size}, '
+                              f'Layers amount={layers}, '
+                              f'Hidden features size={hidden_channels}')
+                        params = {
+                            'inputs_channels': inputs_channels,
+                            'hidden_channels': hidden_channels,
+                            'output_channels': dataset.num_classes,
+                            'num_heads': args.num_heads,
+                            'num_layers': layers,
```

Comment:
If we are standardizing behind `num_neighbors`, here we can just set `'num_layers' = len(num_neighbors)`.